components/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,481 @@
+import {isMac} from 'utils/utils';
+import * as Utils from 'utils/utils';
+
+import DeletePostModal from 'components/delete_post_modal';
+import EmojiPickerOverlay from 'components/emoji_picker/emoji_picker_overlay';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import Textbox, {TextboxClass} from 'components/textbox';
+import EmojiIcon from 'components/widgets/icons/emoji_icon';
+import {isInternetExplorer, isSafari} from '../../utils/user_agent';
+
+type OpenModal = {
+    ModalId: string;
+    dialogType: typeof React.Component;
+    dialogProps: {
+        post: Post;
+        isRHS?: boolean;
+    };
+};
+
+export type Actions = {
+    addMessageIntoHistory: (message: string) => void;
+    editPost: (input: Partial<Post>) => Promise<Post>;
+    unsetEditingPost: () => void;
+    openModal: (input: OpenModal) => void;
+    setShowPreview: (newPreview: boolean) => void;
+    scrollPostListToBottom: () => void;
+}
+
+export type Props = {
+    canEditPost?: boolean;
+    canDeletePost?: boolean;
+    readOnlyChannel?: boolean;
+    channelId: string;
+    codeBlockOnCtrlEnter: boolean;
+    ctrlSend: boolean;
+    config: {
+        EnableEmojiPicker?: string;
+        EnableGifPicker?: string;
+    };
+    maxPostSize: number;
+    shouldShowPreview: boolean;
+    useChannelMentions: boolean;
+    editingPost: {
+        post?: Post;
+        postId?: string;
+        refocusId?: string;
+        title?: string;
+        isRHS?: boolean;
+    };
+    actions: Actions;
+};
+
+export type State = {
+    editText: string;
+    caretPosition: number;
+    postError: React.ReactNode;
+    errorClass: string | null;
+    showEmojiPicker: boolean;
+    renderScrollbar: boolean;
+    scrollbarWidth: number;
+    prevShowState: boolean;
+};
+
+const {KeyCodes} = Constants;
+
+const TOP_OFFSET = 0;
+const RIGHT_OFFSET = 10;
+
+const EditPost = ({editingPost, actions, ...rest}: Props): JSX.Element => {
+    const [editText, setEditText] = useState<string>(
+        editingPost.post?.message_source || editingPost.post?.message || '',
+    );
+    const [caretPosition, setCaretPosition] = useState<number>(0);
+    const [postError, setPostError] = useState<React.ReactNode | null>(null);
+    const [errorClass, setErrorClass] = useState<string>('');
+    const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
+    const [renderScrollbar, setRenderScrollbar] = useState<boolean>(false);
+
+    const textboxRef = useRef<TextboxClass>(null);
+    const emojiButtonRef = useRef<null>(null);
+
+    const {formatMessage} = useIntl();
+
+    const scrollOptions: boolean|ScrollIntoViewOptions = isInternetExplorer() || isSafari() ? false : {behavior: 'smooth', block: 'end'};
+
+    useEffect(() => textboxRef?.current?.focus(), []);
+    useEffect(() => {
+        const parentId = `${editingPost.isRHS ? 'rhsP' : 'p'}ost_${editingPost.postId}`;
```

Comment:
I would do `...'rhsPost' : 'post'}_${...`. We are only saving a few characters, and this way is more readable.